x-pack/plugins/dataset_quality/public/services/data_streams_stats/data_streams_stats_client.ts

```diff
@@ -23,20 +33,51 @@ export class DataStreamsStatsClient implements IDataStreamsStatsClient {
   public async getDataStreamsStats(
     params: GetDataStreamsStatsQuery = { type: 'logs' }
   ): Promise<DataStreamStatServiceResponse> {
-    const { dataStreamsStats, integrations } = await this.http
+    const response = await this.http
       .get<GetDataStreamsStatsResponse>(DATA_STREAMS_STATS_URL, {
         query: params,
       })
       .catch((error) => {
         throw new GetDataStreamsStatsError(`Failed to fetch data streams stats": ${error}`);
       });
 
+    const { dataStreamsStats, integrations } = decodeOrThrow(
+      getDataStreamsStatsResponseRt,
+      (message: string) =>
+        new GetDataStreamsStatsError(`Failed to decode data streams stats response: ${message}"`)
+    )(response);
+
     const mergedDataStreamsStats = dataStreamsStats.map((statsItem) => {
       const integration = find(integrations, { name: statsItem.integration });
 
-      return integration ? merge({}, statsItem, { integration }) : statsItem;
+      return merge({}, statsItem, { integration });
     });
 
     return mergedDataStreamsStats.map(DataStreamStat.create);
   }
+
+  public async getDataStreamsMalformedStats(params: GetDataStreamsMalformedDocsStatsQuery) {
+    const response = await this.http
+      .get<GetDataStreamsMalformedDocsStatsResponse>(DATA_STREAMS_MALFORMED_STATS_URL, {
+        query: {
+          ...params,
+          type: 'logs',
+        },
+      })
+      .catch((error) => {
+        throw new GetDataStreamsStatsError(
```

Comment:
Question - All these errors which we are throwing here, how will they be handled on the UI. Will they appear on a toast ?

Reply:
yes, we have this [issue](https://github.com/elastic/kibana/issues/172156) to create an e2e test suite for dataset quality page and flyout, we need to decide first what we would use, cypress or FTRs.

Reply:
> Question - All these errors which we are throwing here, how will they be handled on the UI. Will they appear on a toast or simply end up in console ?

Couldn't find a toast handler for these ones, I can introduce a handler this in a follow up PR

Reply:
Something we need to handle, giving feedback to users around failures is essential to ensure they know how to change the workflow alone if possible.
In the Log Explorer we handled all the errors at the state machine level and display a toast message with the related insight when they occurred.